src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs

```diff
@@ -0,0 +1,379 @@
+                if (property.Name == "JsonFilePath")
+                {
+                    jsonFilePath = (string)value!;
+                }
+                else
+                    throw new Exception ($"JsonToItemsTask {TaskName} cannot set property {property.Name}");
+            }
+
+        }
+
+        public class JsonModelRoot
+        {
+            [JsonConverter(typeof(CaseInsensitiveDictionaryConverter))]
+            public Dictionary<string, string>?  Properties {get; set;}
+            public Dictionary<string, JsonModelItem[]>? Items {get; set;}
+
+            public JsonModelRoot() {}
+        }
+
+        [JsonConverter(typeof(JsonModelItemConverter))]
+        public class JsonModelItem
+        {
+            public string? Identity {get; set;}
+            // n.b. will  be deserialized case insensitive
+            public Dictionary<string, string>? Metadata {get;  set; }
+        }
+
+        public class CaseInsensitiveDictionaryConverter : JsonConverter<Dictionary<string, string>>
+        {
+            public override Dictionary<string, string> Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                var dict = JsonSerializer.Deserialize<Dictionary<string, string>>(ref reader, options);
+                if (dict == null)
+                    return null!;
+                return new Dictionary<string, string>(dict, StringComparer.OrdinalIgnoreCase);
+            }
+            public override void Write (Utf8JsonWriter writer, Dictionary<string, string>? value, JsonSerializerOptions options) =>
+                JsonSerializer.Serialize(writer, value, options);
+        }
+        public  class JsonModelItemConverter : JsonConverter<JsonModelItem>
+        {
+            public JsonModelItemConverter() {}
+
+            public override JsonModelItem Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                switch (reader.TokenType)
+                {
+                    case JsonTokenType.String:
+                        return new JsonModelItem { Identity = reader.GetString() };
+                    case JsonTokenType.StartObject:
+                        var item = new JsonModelItem();
+                        var dict = JsonSerializer.Deserialize<Dictionary<string, string>>(ref reader, options);
+                        if (dict == null)
+                            return null!;
+                        var idict = new Dictionary<string, string> (dict, StringComparer.OrdinalIgnoreCase);
+                        if  (!idict.TryGetValue("Identity", out var identity) || identity == null)
+                            throw new Exception ("deserialized json dictionary item did not have an Identity metadata");
+                        item.Identity = identity;
+                        item.Metadata = idict;
```

Comment:
Should the `Identity` entry be removed from `idict.Metadata`? We wouldn't need to special case it in the task then

Reply:
Did this and made `JsonModelItem.Identity` non-nullable, which improved things elsewhere. Thanks!